server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java

```diff
@@ -78,8 +78,11 @@ public class SocketServerConfigs {
             " <code>PLAINTEXT://127.0.0.1:9092,SSL://[::1]:9092</code>%n";
 
     public static final String ADVERTISED_LISTENERS_CONFIG = "advertised.listeners";
-    public static final String ADVERTISED_LISTENERS_DOC = String.format(
-            "Listeners to publish to ZooKeeper for clients to use, if different than the <code>%s</code> config property." +
+    public static final String ADVERTISED_LISTENERS_DOC = String.format("Specifies the listener addresses that the Kafka brokers will advertise to clients and other brokers." +
+                    " The config is useful where the actual listener configuration <code>%s</code> does not represent the addresses that clients should" +
+                    " use to connect, such as in cloud environments. In environments using ZooKeeper, these addresses are published to ZooKeeper." +
+                    " In KRaft mode, these information are managed internally by the Kafka brokers themselves with KRaftMetadataCache." +
```

Comment:
Maybe we can simply say `In KRaft mode, these addresses are published to Kraft controller`?